src/forestTypes/binnedTree/binnedBase.h

```diff
@@ -42,26 +44,24 @@ namespace fp {
 			binnedBase(){
 				checkParameters();
 				numBins =  fpSingleton::getSingleton().returnNumTreeBins();
-				nodeIndices.resize(fpSingleton::getSingleton().returnNumObservations());
-				for(int i = 0; i < fpSingleton::getSingleton().returnNumObservations(); ++i){
-					nodeIndices[i] =i;
-				}
+				generateSeedsForBins();
 			}
 
-			fpDisplayProgress printProgress;
+			inline void generateSeedsForBins(){
+				binSeeds.resize(numBins);
+				for(int i = 0; i < numBins; ++i){
+					binSeeds[i] = fpSingleton::getSingleton().genRandom(std::numeric_limits<int>::max());
```

Comment:
This was replaced above by `randNum.gen()`?

Reply:
This will need to change.  Right now each bin gets seeded and the bin creates seeds for each of their trees.  This is problematic because reproducibility is only possible if the same number of trees and bins are chosen.  A better way to do reproducibility is to generate a seed for each of the trees then seed the tree when it first starts growing.  This way you always get the same forest every time the seed and number of trees match... although other parameters may also affect this.